### DISCUSSION TITLE
Help - How do I upload file size that is more than 15 mb in size to create knowledge?

### DISCUSSION THREAD

**Discussion by murtaza-motorwala** [langgenius/dify#11072] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

How do I upload file size that is more than 15 mb in size to create knowledge? Is there a place or setting in docker-compose.yaml which I can change to do so?

**Comment by zolgear** [comment]
https://docs.dify.ai/getting-started/install-self-hosted/environments#knowledge-configuration

Please update the `.env` file with the following variables:  

- `UPLOAD_FILE_SIZE_LIMIT`  
- `NGINX_CLIENT_MAX_BODY_SIZE`  

If you encounter an nginx error in the container logs during file upload, increase the `NGINX_CLIENT_MAX_BODY_SIZE` to match the desired value.
